### ISSUE TITLE
Docker `armhf` deprecation

### ISSUE THREAD

**Issue by BlaineEXE** [rook/rook#1419]
There is a deprecation warning in the `armhf` org on DockerHub.

> The armhf organization is deprecated in favor of the more-specific arm32v7 and arm32v6 organizations, as per https://github.com/docker-library/official-images#architectures-other-than-amd64. Please adjust your usages accordingly.

It seems that it might make sense to change the arm builds (see `images/image.mk`) to:
`arm32v7` and `arm64v8`. 

I'm not sure if adding `arm32v5` or `arm32v6` also makes sense. It depends on what platforms Rook wants to target. v6 is R-Pi FWIW. I wouldn't think that R-Pi needs to be supported, but some people do want to run clusters of R-Pis for fun. I don't see any good reason to support v5.

**Comment by BlaineEXE** [comment]
I'm still trying to make sure I understand how the builds work. I'm a bit confused. Is the `images/image.mk` file used as part of the rook build? It seems like this file could be for/by your Jenkins setup?

**Comment by bassam** [comment]
@BlaineEXE agree that we should use `arm64v8` base images going forward. `rook/rook` images are multi-arch and built using [manifest-tool](https://github.com/estesp/manifest-tool), so that would need to altered too (at a minimum to set the v8 variation).

**Comment by BlaineEXE** [comment]
I thought I could look into making these changes myself, and I started by making sure I could do the image builds as-is, and I've been running into issues.

I've cd'ed to the `images/` dir, and I am executing `make`. Are there any steps I'm missing? It seems as thought the `_output/` dir isn't getting created.

**Comment by BlaineEXE** [comment]
I'm also a little curious about why the docker builds are so complex. The rook container has both a .base and a regular dockerfile, as does the toolbox. And the base container is created based, then its filesystem is exported to another scratch container, and then that is used as a base to build the official base. It seems like it would be cleaner for some of these to be merged into one Dockerfile, but I imagine there is some purpose served by the complexity that I don't see.

**Comment by bassam** [comment]
I would just build from the root dir using `make build.all` which builds all platforms.

the docker images are more complex than usual due to the fact that we were building ceph from source inside containers, leveraging docker image caching for that, and supporting things like ccache and apt-caching. As of #1362 we moved away from this approach and I did a first take on simplifying. Its likely due for another round of simplification similar to the work that went into https://github.com/ceph/ceph-container/pull/878 recently.
